core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -96,14 +79,12 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
             filter.toXContent(builder, params);
         }
         builder.endArray();
-        if (queryName != null) {
-            builder.field("_name", queryName);
-        }
+        printBoostAndQueryName(builder);
```

Comment:
Here you call super method while in toQuery/doToQuery, doEquals, doWrite etc... superclass calls abstract method. Any reason why this is different here? Might be more consistent to follow one pattern and have the superclass always call concrete implementation? Just an idea really.